src/lib/swapper/swappers/ZrxSwapper/zrxExecuteTrade/zrxExecuteTrade.ts

```diff
@@ -2,26 +2,37 @@ import type { Result } from '@sniptt/monads'
 import { Err, Ok } from '@sniptt/monads'
 import type { SwapErrorRight, TradeResult } from 'lib/swapper/api'
 import { makeSwapErrorRight, SwapError, SwapErrorType } from 'lib/swapper/api'
-import { buildAndBroadcast, isNativeEvmAsset } from 'lib/swapper/swappers/utils/helpers/helpers'
+import {
+  buildAndBroadcast,
+  createBuildCustomTxInput,
+  isNativeEvmAsset,
+} from 'lib/swapper/swappers/utils/helpers/helpers'
 import type { ZrxExecuteTradeInput, ZrxSwapperDeps } from 'lib/swapper/swappers/ZrxSwapper/types'
 import type { ZrxSupportedChainId } from 'lib/swapper/swappers/ZrxSwapper/ZrxSwapper'
 
 export async function zrxExecuteTrade<T extends ZrxSupportedChainId>(
   { adapter }: ZrxSwapperDeps,
   { trade, wallet }: ZrxExecuteTradeInput<T>,
 ): Promise<Result<TradeResult, SwapErrorRight>> {
-  const { accountNumber, depositAddress, feeData, sellAsset, txData } = trade
-  const { sellAmountBeforeFeesCryptoBaseUnit } = trade
+  const { accountNumber, depositAddress, sellAmountBeforeFeesCryptoBaseUnit, sellAsset, txData } =
+    trade
 
   try {
-    const txid = await buildAndBroadcast({
+    const buildCustomTxArgs = await createBuildCustomTxInput({
       accountNumber,
       adapter,
-      feeData: feeData.chainSpecific,
-      to: depositAddress,
-      value: isNativeEvmAsset(sellAsset.assetId) ? sellAmountBeforeFeesCryptoBaseUnit : '0',
+      erc20ContractAddress: depositAddress,
+      erc20ContractData: txData,
+      sendAmountCryptoBaseUnit: isNativeEvmAsset(sellAsset.assetId)
```

Comment:
Not an ERC-20 - according to ZRX docs, `to` (which we alias to `depositAddress` is `The address of the contract to send call data to.`) meaning this is indeed a contract address, but not an ERC-20. In the case of ZRX, that's [`ZeroEx.sol`](https://docs.0xprotocol.org/en/latest/architecture/proxy.html?highlight=zeroex#proxy) i.e the proxy contract executing trades.

Let's probably disregard my [previous comment](https://github.com/shapeshift/web/pull/4492#discussion_r1195522985) WRT `getApprovalContractData`'s `erc20ContractAddress` param and consistently call this one `to`?

- in the case of an approval, we're calling the contract of the asset being approved, the address of which is the `to`
- in the case of a trade execution, we're also sending some calldata to a `to`, though not an ERC-20

Reply:
And WRT `sendAmountCryptoBaseUnit` I reckon we can keep the previous `value` terminology, which is honoring chain-adapters/the Ethereum JSON-RPC spec i.e implied to be a base unit native asset amount

Reply:
good call, i've reverted the naming of all of this to the standard vernacular :)